mailmine.py

```diff
@@ -67,6 +68,11 @@
         except phonenumbers.NumberParseException:
             pass
 
+
+counters = collections.defaultdict(collections.Counter)
+counters[email['From']] += 1
+counters[email['From']] [date.hour]+= 1
```

Comment:
This should also be a part of the main loop, currently you have it outside the loop so no details of the individual messages are available as variables here.